go.mod

```diff
@@ -20,8 +20,8 @@ require (
 	github.com/google/go-github/v45 v45.2.0
 	github.com/google/gofuzz v1.2.0
 	github.com/mattn/go-runewidth v0.0.13 // indirect
-	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.18.1
+	github.com/onsi/ginkgo/v2 v2.1.4
+	github.com/onsi/gomega v1.19.0
```

Comment:
Usually we keep those versions in sync with CR. (https://github.com/kubernetes-sigs/controller-runtime/blob/master/go.mod)

Should we make an exception in this case? 

I think we don't really depend on using the same ginkgo version as CR. It's also a /v2 dependency so from
a go module perspective it's kind of another dependency and we could still important the same v1 as CR if that becomes necessary.

@vincepri WDYT?

Reply:
If we update ginkgo to v2 in capz @ main then we won't be able to import capi 1.2 without this change landing in 1.2

Reply:
> Should we make an exception in this case?

I'm not necessarily arguing for that. Maybe the right thing to do is update controller-runtime to Ginkgo v2 first.

Reply:
> If we update ginkgo to v2 in capz @ main...

I think updating Ginkgo in provider projects is effectively blocked until this merges. You can't import v1 and v2 at the same time since they'll conflict in handling [flags](https://github.com/onsi/ginkgo/issues/875#issuecomment-1036350659).

Reply:
Forgot: cc @fabriziopandini